microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceDiagnosticsParticipant.java

```diff
@@ -0,0 +1,113 @@
+import static org.eclipse.lsp4mp.jdt.internal.faulttolerance.java.MicroProfileFaultToleranceErrorCode.FALLBACK_METHOD_DOES_NOT_EXIST;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IMemberValuePair;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.ITypeRoot;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.IJavaDiagnosticsParticipant;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.JavaDiagnosticsContext;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTTypeUtils;
+import org.eclipse.lsp4mp.jdt.core.utils.PositionUtils;
+
+public class MicroProfileFaultToleranceDiagnosticsParticipant implements IJavaDiagnosticsParticipant {
+
+	private static final Pattern fallbackMethodNameExtractor = Pattern.compile(FALLBACK_METHOD_FALLBACK_ANNOTATION_MEMBER + "\\s*=\\s*([^,]+),");
+	
+	@Override
+	public boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		IJavaProject javaProject = context.getJavaProject();
+		return JDTTypeUtils.findType(javaProject, FALLBACK_ANNOTATION) != null;
+	}
+
+	@Override
+	public List<Diagnostic> collectDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		ITypeRoot root = context.getTypeRoot();
+		IJavaElement[] elements = root.getChildren();
+		List<Diagnostic> diagnostics = new ArrayList<>();
+		collectDiagnostics(elements, diagnostics, context, monitor);
+		return diagnostics;
+	}
+
+	private void collectDiagnostics(IJavaElement[] elements, List<Diagnostic> diagnostics,
+			JavaDiagnosticsContext context, IProgressMonitor monitor) {
+		for (IJavaElement element : elements) {
+			if (monitor.isCanceled()) {
+				return;
+			} else if (element.getElementType() == IJavaElement.TYPE) {
+				IType type = (IType) element;
+				validateClass(type, diagnostics, context, monitor);
+			}
+		}
+	}
+
+	private void validateClass(IType type, List<Diagnostic> diagnostics, JavaDiagnosticsContext context,
+			IProgressMonitor monitor) {
+		try {
+			IMethod[] methods = type.getMethods();
+			Set<String> methodSet = Stream.of(methods).map(m -> {
+				return m.getElementName();
+			}).collect(Collectors.toUnmodifiableSet());
+			for (IMethod method : methods) {
+				if (monitor.isCanceled()) {
+					return;
+				}
+				IAnnotation annotation = method.getAnnotation(FALLBACK_ANNOTATION);
+				if (annotation != null && annotation.exists()) {
```

Comment:
`annotation.exists()` returns false when I run this code, so no validation is provided. If I don't do this check, I instead get a JavaModelException when running `getFallbackName()`.